packages/ses/test/test-freeze-module-namespace.js

```diff
@@ -0,0 +1,186 @@
+  t.is(descs1.three.writable, true, 'const three is writable');
+  t.is(descs1.four.writable, true, 'const four is writable');
+
+  // Surprising that each freeze actually freezes only one property and then
+  // fails.
+
+  t.throws(() => freeze(nsConst34), {
+    message: /Cannot assign to read only property 'four' of object '\[object Module\]'/,
+  });
+  t.is(isFrozen(nsConst34), false, 'still not frozen');
+  const descs2 = getOwnPropertyDescriptors(nsConst34);
+  t.is(descs2.four.writable, false, 'const four no longer writable');
+  t.is(descs2.three.writable, true, 'const three still writable');
+  // console.log(JSON.stringify(descs2));
+
+  t.throws(() => freeze(nsConst34), {
+    message: /Cannot assign to read only property 'three' of object '\[object Module\]'/,
+  });
+  t.is(isFrozen(nsConst34), true, 'now it is frozen');
+  const descs3 = getOwnPropertyDescriptors(nsConst34);
+  t.is(descs3.four.writable, false, 'const four no longer writable');
+  t.is(descs3.three.writable, false, 'const three no longer writable');
+  // console.log(JSON.stringify(descs3));
+});
+
+test('freeze module-namespace-live-5-6', t => {
+  t.is(isFrozen(nsLive56), false, 'not pre-frozen');
+  t.is(isExtensible(nsLive56), false, 'is pre non-extensible');
+  const descs1 = getOwnPropertyDescriptors(nsLive56);
+  t.is(descs1.writeFive.writable, true, 'let writeFive is writable');
+  t.is(descs1.five.writable, true, 'const five is writable');
+
+  // Slightly surpring that `four` was frozen before `three` above but
+  // `writeFive` (which is textually second, like `four` was) is frozen
+  // after `five`. But the bug surprise is explained below.
+
+  t.throws(() => freeze(nsLive56), {
+    message: /Cannot assign to read only property 'five' of object '\[object Module\]'/,
+  });
+  t.is(isFrozen(nsLive56), false, 'still not frozen');
+  const descs2 = getOwnPropertyDescriptors(nsLive56);
+  t.is(descs2.five.writable, false, 'five no longer writable');
+  t.is(descs2.writeFive.writable, true, 'let writeFive still writable');
+  // console.log(JSON.stringify(descs2));
+
+  t.throws(() => freeze(nsLive56), {
+    message: /Cannot assign to read only property 'writeFive' of object '\[object Module\]'/,
+  });
+  t.is(isFrozen(nsLive56), true, 'now it is frozen');
+  const descs3 = getOwnPropertyDescriptors(nsLive56);
+  t.is(descs3.five.writable, false, 'five no longer writable');
+  t.is(descs3.writeFive.writable, false, 'let writeFive no longer writable');
+  // console.log(JSON.stringify(descs3));
+
+  t.is(typeof nsLive56.writeFive, 'function');
+  nsLive56.writeFive();
+
+  const descs4 = getOwnPropertyDescriptors(nsLive56);
+
+  // OMG Surprising that a fundamental object invariant was violated!
+  // The `nsLive.five` property was observed to be a
+  // non-writable non-configurable data property with value 5.
+  // This same property was then observed to be a
+  // non-writable non-configurable data property with value 6.
```

Comment:
This is the scary part I suspected when you wrote the initial isFrozen test.

Reply:
Y. This definitely demonstrates a spec violation. But we'd rather they fixed this the more pleasant way by "fixing" the spec at the same time.